packages/spec-lib/lib/lib.tsp

```diff
@@ -17,7 +17,7 @@ alias ScenarioServiceOptions = {
 extern dec scenarioService(
   target: Namespace,
   route: valueof string,
-  options?: valueof ScenarioServiceOptions
+  options?: ScenarioServiceOptions
```

Comment:
why is this changed too?

Reply:
@timotheeguerin This change is required for the following reason:

In the `packages/azure-http-specs/specs/azure/core/model/main.tsp` file, we have the following decorator code:

```
@scenarioService(
  "/azure/core/model",
  {
    versioned: Versions,
  }
)
```

If I have the `valueof` qualifier in the `spec-lib` repository, this will give the following error during the build (of the `azure-http-specs` package)

```
error: ✘ Argument of type '{ versioned: _Specs_.Azure.Core.Model.Versions }' is not assignable to parameter of type 'valueof TypeSpec.SpecLib.{ versioned: Enum }': C:/Users/sarajama/Projects/typespec-azure/packages/azure-http-specs/specs/azure/core/model/main.tsp:15:3
```

**How is this scenario handled in cadl-ranch repository?**

The `cadl-ranch-expect` package does not have the `valueof` qualifier. So, there is no problem there.